### PR TITLE
Added relevant and go-compatable tests to testing file. Fixed dead PID on windows.

### PR DESCRIPTION
**Pull request by stewi1014**
This is my contribution. I've been pretty radical with my fixes despite my attempts at being conservative. If it's too radical that's fine, if I've done something wrong please tell me.

(Finally tested on Linux)

### CODE REVIEW

**Comment by xiegeo** [comment]
@stewi1014 Thank you for this. I just thought you weren't going to push your changes so I sent my own. All the better that there are now two to choose from.

**Comment by stewi1014** [comment]
@xiegeo Yeh, my bad. I'm sorry about saying I'd go and make a fix for it and then kind of disappearing for a week. That was bad form on my part.

File: lockfile_test.go [hunk 1]
```diff
@@ -1,28 +1,112 @@
...
 	"fmt"
-	"os"
+	"io/ioutil"
+	"math/rand"
 	"path/filepath"
+	"strconv"
+	"testing"
 )
 
-func ExampleLockfile() {
```
Comment (nightlyone) [inline lockfile_test.go]: Could you keep the example? I consider an automatically verified example that can be copy'n'pasted as rather important for a package.

File: lockfile_windows.go [hunk 2]
```diff
@@ -1,32 +1,24 @@
...
-	handle := syscall.Handle(field.Uint())
-
-	var code uint32
-	err := syscall.GetExitCodeProcess(handle, &code)
+func isRunning(pid int) (bool, error) {
+	procHnd, err := syscall.OpenProcess(syscall.PROCESS_QUERY_INFORMATION, true, uint32(pid))
 	if err != nil {
-		return err
+		if scerr, ok := err.(syscall.Errno); ok {
+			if uintptr(scerr) == 87 {
```
Comment (nightlyone) [inline lockfile_windows.go]: Could you use a symbolic constant instead of a magic value here?

File: lockfile_windows.go [hunk 3]
```diff
@@ -1,32 +1,24 @@
...
-	if code == 259 {
-		return nil
+	var code uint32
+	err = syscall.GetExitCodeProcess(procHnd, &code)
+	if err != nil {
+		return false, err
 	}
 
-	return ErrDeadOwner
+	return code == 259, nil
```
Comment (nightlyone) [inline lockfile_windows.go]: Could you use a symbolic constant instead of a magic value here? Or at least keep the comment.

**Comment by nightlyone** [comment]
Thanks for your great work @stewi1014 and @xiegeo ! I added a few comments, that would make me happy to merge this, if resolved.

File: lockfile_test.go (hunk 1 above)
Comment (stewi1014) [inline lockfile_test.go]: I'm afraid I don't really know what you're talking about. I have tried following the documentation on writing test functions for the library (https://golang.org/pkg/testing/) as much as I can.

I'll put the old test file back, but I'm struggling so see the reasoning behind deviating form the go standard. Am I missing something?

**Comment by stewi1014** [comment]
Just letting you know I've committed the changes.

**Comment by nightlyone** [comment]
@stewi1014 awesome! this looks great now!

**Pull request merged**